pkg/config/config_factory.go

```diff
@@ -34,10 +36,11 @@ func GetConfigProvider() service.ConfigProvider {
 	if configContent, ok := os.LookupEnv(envKey); ok {
 		log.Printf("Reading AOT config from environment: %v\n", configContent)
 		return service.NewConfigProvider(
-			[]configmapprovider.Provider{configmapprovider.NewEnv(envKey)},
-			[]service.ConfigMapConverterFunc{NewExpandConverter()},
+			[]string{fmt.Sprintf("env:%s", envKey)},
+			map[string]configmapprovider.Provider{"env": configmapprovider.NewEnv()},
+			[]config.MapConverterFunc{configmapprovider.NewExpandConverter()},
```

Comment:
The `env:` scheme provider should now be [included in the default config provider](https://github.com/open-telemetry/opentelemetry-collector/blob/main/service/config_provider.go#L109-L110), so this can be collapsed to determining which location to provide to `NewDefaultConfigProvider()`:

```go
loc := getConfigFlag()
if configContent, ok := os.LookupEnv(envKey); ok {
	log.Printf("Reading AOT config from environment: %v\n", configContent)
	loc = "env:" + envKey
}

return service.NewDefaultConfigProvider([]string{loc}, getSetFlag())
```

Reply:
Thank you for the clarification! Resolved.